apps/wallet/src/ui/app/components/network-selector/custom-rpc-input/index.tsx

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import cl from 'classnames';
+import { Field, Formik, Form } from 'formik';
+import { useState, useCallback } from 'react';
+import * as Yup from 'yup';
+
+import { ENV_TO_API } from '_app/ApiProvider';
+import Button from '_app/shared/button';
+import InputWithAction from '_app/shared/input-with-action';
+import Alert from '_components/alert';
+import { useAppSelector, useAppDispatch } from '_hooks';
+import { setCustomRPC } from '_redux/slices/app';
+
+import st from '../NetworkSelector.module.scss';
+
+const MIN_CHAR = 5;
+
+const validation = Yup.object({
+    rpcInput: Yup.string()
+        .required()
+        .label('Custom RPC URL')
+        .min(MIN_CHAR)
+        .test(
+            'validate-url',
+            'Not a valid URL',
+            (value) =>
+                value?.indexOf('http://') === 0 ||
```

Comment:
We could use `new URL()` with try catch and return false on error. `http(s)://` is not a valid url either